x-pack/plugins/ml/public/application/timeseriesexplorer/components/entity_control/entity_control.tsx

```diff
@@ -136,6 +199,132 @@ export class EntityControl extends Component<EntityControlProps, EntityControlSt
         isClearable={false}
         renderOption={this.renderOption}
         data-test-subj={`mlSingleMetricViewerEntitySelection ${entity.fieldName}`}
+        prepend={
+          <EuiPopover
+            ownFocus
+            button={
+              <EuiButtonIcon
+                color="text"
+                iconSize="xxl"
+                iconType="gear"
+                aria-label={i18n.translate('xpack.ml.timeSeriesExplorer.editControlConfiguration', {
+                  defaultMessage: 'Edit field configuration',
+                })}
+                onClick={() => {
+                  this.setState({
+                    isEntityConfigPopoverOpen: !this.state.isEntityConfigPopoverOpen,
+                  });
+                }}
+                data-test-subj={`mlSingleMetricViewerEntitySelectionConfigButton_${entity.fieldName}`}
+              />
+            }
+            isOpen={this.state.isEntityConfigPopoverOpen}
+            closePopover={() => {
+              this.setState({
+                isEntityConfigPopoverOpen: false,
+              });
+            }}
+          >
+            <EuiText>
+              <EuiFlexGroup gutterSize={'xs'} alignItems={'center'}>
+                <EuiFlexItem grow={false}>
+                  <EuiSwitch
+                    label={
+                      <FormattedMessage
+                        id="xpack.ml.timeSeriesExplorer.anomalousOnlyLabel"
+                        defaultMessage="Anomalous only"
+                      />
+                    }
+                    checked={!!this.props.config?.anomalousOnly}
+                    onChange={(e) => {
+                      const isAnomalousOnly = e.target.checked;
+                      this.props.onConfigChange(this.props.entity.fieldType, {
+                        anomalousOnly: isAnomalousOnly,
+                        sort: {
+                          order: this.props.config.sort.order,
+                          by: this.props.config.sort.by,
+                        },
+                      });
+                    }}
+                    compressed
+                    data-test-subj={`mlSingleMetricViewerEntitySelectionConfigAnomalousOnly_${entity.fieldName}`}
+                  />
+                </EuiFlexItem>
+
+                <EuiFlexItem grow={false} style={{ width: '16px', height: '24px' }}>
+                  {!this.props.config?.anomalousOnly ? (
+                    <EuiToolTip
+                      position="top"
+                      content={
+                        this.props.isModelPlotEnabled ? (
+                          <FormattedMessage
+                            id="xpack.ml.timeSeriesExplorer.nonAnomalousResultsWithModelPlotInfo"
+                            defaultMessage="You will be suggested to select values for all possible model plot results"
```

Comment:
Likewise, I think this can be changed to an informational message like this:
```suggestion
                            defaultMessage="The list contains values for all possible model plot results."
```
... though I'm still not sure that I understand what the difference is between the model plot results and all records.

Reply:
Changed in https://github.com/elastic/kibana/pull/81510/commits/79d10e7bfcf867fd204ded3a31d1265bf3bd0788

Reply:
What about `Displays a list of all entities that have been modeled by this job, including those that have not generated anomalies.` to avoid low level details of what type of results are being queried in this case?

Reply:
`including those that have not generated anomalies` this is not strictly accurate. It would have to be something like `including those that have not recently generated anomalies`

Reply:
We decided against searching the raw data so it will be all entities if model plot is enabled, but otherwise it is things which have at one time generated an anomaly.

Reply:
Sorry my mistake. I didn't realise we had two tooltips one for each case. Ignore my comments

Reply:
So for the case where model plot is enabled, maybe `Displays a list of all entities that have been modeled by this job, including those that have not generated anomalies during the selected time range.` ?

Reply:
I'm still confused by this description, so in case it's not just me, I'll describe here. In the model-plot-disabled case, it sounds like the list contains all the entities from all the "get records" API results all the way back to the beginning of the job. In the model-plot-enabled case, it sounds like it's a subset of entities that "have model plot results" or "have model plot information" or "have been modeled by this job" ... but those descriptions don't align with anything concrete for me.  Is there some field that is present or absent from an API result for a particular record that indicates it has model plot information? I've never seen what model_plot does under the covers, just that it makes the bounds visible... so as-is these descriptions are quite abstract for me. If you want to talk through these concepts and try to come up with something more tangible, I'm happy to meet.

Reply:
Thanks for the offline explanations. I've created an issue to augment the docs with more information about the different types of results (and interpretation of anomaly detection results in general) in https://github.com/elastic/stack-docs/issues/1430